domains/Links/Tests/Feature/LinksStoreLimitTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace Domains\Links\Tests\Feature;
+
+use Domains\Tags\Database\Factories\TagFactory;
+use Domains\Tags\Models\Tag;
+use Faker\Factory;
+use Faker\Generator;
+use Illuminate\Http\Response;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Http;
+use Illuminate\Support\Facades\Storage;
+use Laravel\Lumen\Testing\DatabaseMigrations;
+use Tests\TestCase;
+
+class LinksStoreLimitTest extends TestCase
```

Comment:
This is a good solution. However, I'd argue that since the check is happening on the model observer layer, we don't need to use the API endpoints.

We could expect the `UnapprovedLinkLimitException`, create 5 Link resources related to the same author email, with the factories (maybe on the `setUp()` method of the test class), and then try to create a 6th which should throw the exception and we could be ensured that whatever the context, the observer is doing its job.

What do you think?

Reply:
Agree.